sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_sort.asciidoc

```diff
@@ -0,0 +1,325 @@
+|===
+
+Table. Constructors of the `default_sorter` class.
+|===
+|Constructor|Description
+
+|`default_sorter(Compare comp = Compare())`
+|Creates the `default_sorter` object using `comp`.
+|===
+
+Table. Member functions of the `default_sorter` class.
+|===
+|Member function|Description
+
+|`template<typename Group, typename Ptr>
+void operator()(Group g, Ptr begin, Ptr end)`
+|Implements a default sorting algorithm. It's callable by the `joint_sort` algorithm.
+
+_Complexity_: Let `N` be `end - begin`. `O(Nlog^2(N))` comparisons.
+
+|`template<typename Group, typename T>
+T operator()(Group g, T val)`
+|Implements a default sorting algorithm. It's callable by the `sort_over_group` algorithm.
+
+_Complexity_: Let `N` be the work group size. `O(Nlog^2(N))` comparisons.
+|===
+
+Table. Constructors of the `radix_sorter` class.
+|===
+|Constructor|Description
+
+|`radix_sorter(const std::bitset<sizeof(T) * CHAR_BIT> mask = std::bitset<sizeof(T) * CHAR_BIT>
+(std::numeric_limits<unsigned long long>::max()));`
+|Creates the `radix_sorter` object to sort values considering only bits
+that corresponds to 1 in `mask`.
+|===
+
+Table. Member functions of the `radix_sorter` class.
+|===
+|Member function|Description
+
+|`template<typename Group, typename Ptr>
+void operator()(Group g, Ptr begin, Ptr end)`
+|Implements the radix sort algorithm. It's callable by the `joint_sort` algorithm.
+
+|`template<typename Group, typename T>
+T operator()(Group g, T val)`
+|Implements the radix sort algorithm. It's callable by the `sort_over_group` algorithm.
+|===
+
+==== Sort
+The sort function from the {cpp} standard sorts elements with respect to
+the binary comparison function object.
+
+SYCL provides two similar algorithms:
+
+`joint_sort` uses the work-items in a group to execute the corresponding
+algorithm in parallel.
+
+`sort_over_group` performs a sort over values held directly by the work-items
+in a group, and results returned to work-item `i` represent values that are in
+position `i` in the ordered range.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+  template <typename Group, typename Ptr>
+  void joint_sort(Group g, Ptr first, Ptr last); // (1)
+
+  template <typename Group, typename Ptr, typename Compare>
+  void joint_sort(Group g, Ptr first, Ptr last, Compare comp); // (2)
+
+  template <typename Group, typename Ptr, typename Sorter>
+  void joint_sort(Group g, Ptr first, Ptr last, Sorter sorter); // (3)
+
+  template <typename Group, typename T>
+  T sort_over_group(Group g, T val); // (4)
+
+  template <typename Group, typename T, typename Compare>
+  T sort_over_group(Group g, T val, Compare comp); // (5)
+
+  template <typename Group, typename T, typename Sorter>
+  T sort_over_group(Group g, T val, Sorter sorter); // (6)
+}
+----
+
+_Constraints_: All functions are available only if `sycl::is_group_v<std::decay_t<Group>>`
+is true and `sycl::ext::oneapi::is_sorter_v<std::decay_t<Sorter>>` is true.

```

Comment:
I thought we decided to get rid of `is_sorter`?

Reply:
Correct. It's just an atavism.